mlir/lib/Dialect/Affine/Utils/LoopUtils.cpp

```diff
@@ -1406,8 +1406,8 @@ unsigned mlir::affine::permuteLoops(MutableArrayRef<AffineForOp> input,
   // Move the innermost loop body to the loop that would be the innermost in the
   // permuted nest (only if the innermost loop is going to change).
   if (permMap.back() != input.size() - 1) {
-    auto *destBody = input[invPermMap.back().second].getBody();
-    auto *srcBody = input.back().getBody();
+    Block *destBody = ((AffineForOp)input[invPermMap.back().second]).getBody();
```

Comment:
Why add the C style casts?

Reply:
I feel like `MutableArrayRef` is probably correct given it's mutating the `AffineForOp`s... :thinking:

Reply:
It doesn't mutate the _value_ in the array, so `ArrayRef` is fine. Not a fan of C-style casts, but okay.

Reply:
The C-style cast is to work around const'ness friction in MLIR - MLIR IR types don't use `const`s but ArrayRef and similar LLVM data structures will give you `const`.

There's no `AffineForOp` mutation happening here.